src/pycountry/tests/test_general.py

```diff
@@ -17,9 +18,32 @@ def logging():
     logging.basicConfig(level=logging.DEBUG)
 
 
+@pytest.mark.parametrize(
+    "obj, db_type, db_factory",
+    [
+        (pycountry.countries, pycountry.ExistingCountries, pycountry.Country),
+        (pycountry.currencies, pycountry.Currencies, pycountry.Currency),
+        (pycountry.languages, pycountry.Languages, pycountry.Language),
+        (pycountry.scripts, pycountry.Scripts, pycountry.Script),
+        (
+            pycountry.subdivisions,
+            pycountry.Subdivisions,
+            pycountry.SubdivisionHierarchy,
+        ),
+    ],
+)
+def test_expected_instance_types(
+    obj: pycountry.db.Database,
+    db_type: Type[pycountry.db.Database],
+    db_factory: Type[pycountry.db.Data],
+) -> None:
+    assert isinstance(obj, db_type)
+    assert db_type.factory is db_factory
+    assert isinstance(next(iter(obj)), db_factory)
+
+
 def test_country_list():
     assert len(pycountry.countries) == 249
-    assert isinstance(list(pycountry.countries)[0], pycountry.db.Data)
```

Comment:
These are redundant and are wider than the updated `test_expected_instance_types` so I'm letting that test handle the type checking and these handle the counts or other attribute checks